src/metabase/models/table.clj

```diff
@@ -62,9 +62,10 @@
                   :field_order  (driver/default-field-order (t2/select-one-fn :engine Database :id (:db_id table)))}]
     (merge defaults table)))
 
+;; TODO: should we do this after the deletion?
```

Comment:
Before is fine, and it's standard to do this cleanup before deletion for other models